6MI0800193/Week1/Task3.cpp

```diff
@@ -0,0 +1,103 @@
+#include <iostream>
+using namespace std;
+enum class Year {
+    first=1,
+    second=2,
+    third=3,
+    fourth=4
+};
+
+struct Student
+{
+    char name[16];
+    unsigned int fn;
+    Year school_year;
+    double grade;
+};
+
+void print_student(const Student& student)
+{
+    cout << student.name << " ";
+    cout << student.fn << " ";
+    switch (student.school_year)
+    {
+    case Year::first:
+        cout << "first" << " ";
+        break;
+    case Year::second:
+        cout << "second" << " ";
+        break;
+    case Year::third:
+        cout << "third" << " ";
+        break;
+    case Year::fourth:
+        cout << "fourth" << " ";
+        break;
+    }
+
+    cout << student.grade << endl;
+}
+
+bool init_student(Student& toInit, char name[16], unsigned int facNumber, Year currentCourse, double grade)

```

Comment:
This hasn't really been covered yet (exceptions & error handling) but having a boolean return value to signal the success / failure of an operation (in this case - the initialization of an object) is not a good idea. Boolean values don't really mean much (if false nobody will be able to tell the tale of what exactly went wrong) and can easily be overlooked (e.g. if I don't do an explicit check about the given return value - I'll never now that an error occurred). Keep in mind that [some other mechanisms](https://www.tutorialspoint.com/cplusplus/cpp_exceptions_handling.htm) exist for doing just that :))